### ISSUE TITLE
Add option to change the nk3 secret apps pin for reowner ship

### ISSUE THREAD

**Issue by nestire** [Nitrokey/nitrokey-hotp-verification#40]
see also #36

this is not yet possible in heads only via nitropy or the nitrokey app 2

**Comment by tlaurion** [comment]
GPG Admin/User PINs are reset through gpg factory reset under Heads. If Admin PIN has no counter left then Admin is locked out and factory reset needs to be done there. That was also the case for HOTP in nk2/librem key. 

The bare minimal here would be to have hotp-verification permit factory reset of the secret app just like gpg does for opengpg smartcard.


No issue was opened to tackle the issue of physical presence. Could this be optionally turned off when reset? This prevent unattended oem factory reset and transfer of ownership improvements to happen and I see no justification for touch requirement in remote attestation use case but maybe for servers: PIN does the physical presence + authentication.

**Comment by nestire** [comment]
@tlaurion  for the presence see #41 but not sure if there is a good way of doing this

**Comment by daringer** [comment]
this will lead to a new command line argument: `change-pin <old-pin> <new-pin>`

how to handle backwards compatibility? will this just fail for pro/storage? 
I don't think it makes sense to re-implement gpg functionality here.

**Comment by tlaurion** [comment]
> this will lead to a new command line argument: `change-pin <old-pin> <new-pin>`
> 
> how to handle backwards compatibility? will this just fail for pro/storage? 
> I don't think it makes sense to re-implement gpg functionality here.

@JonathonHall-Purism comments?

**Comment by daringer** [comment]
> I'm not sure I understand why simple equivalent of nitropy nk3 secret reset is not implemented first, with Hotp-verification being transparent when it sets a pin to secret element, permitting heads to set it as part of ownership, re-ownership

> PIN change would be nice, but it's not the blocker here. HOTP sealing, resealing and PIN provisioning is the blocker @daringer

please don't derail this issue, this has nothing to do with reset - we can for sure prioritize reset - no problem, this issue - as you requested - is about something else. If priorities belong somewhere, then into the parent issue where you asked for these issues.

**Comment by tlaurion** [comment]
> this will lead to a new command line argument: `change-pin <old-pin> <new-pin>`
> 
> how to handle backwards compatibility? will this just fail for pro/storage? 
> I don't think it makes sense to re-implement gpg functionality here.

@daringer @JonathonHall-Purism in my opinion that wouldn't break anyrjing since it was not possible to do so before?

**Comment by daringer** [comment]
ok, summarizing what shall be implemented:
* introduce cmdline option: `change-pin <old-pin> <new-pin>`
* this shall change the secrets pin (or set if not set) on a nitrokey 3
* this operation does nothing if a pro2/storage is connected (silent fail w/o bad return error code)

**Comment by tlaurion** [comment]
@JonathonHall-Purism fyi

**Comment by JonathonHall-Purism** [comment]
This sounds like it should work.  Thanks @daringer

**Comment by tlaurion** [comment]
Ideal is still for hotp_verification reset SECRET_APP_PIN

**Comment by tlaurion** [comment]
Ideal is still for hotp_verification reset SECRET_APP_PIN gpg requires a change of pin because factory reset sets it to something useless in most cases and heads then have to sldona second round of ops.


I said #40 was nice to have since the beginning.
Previous comment under #42

**Comment by daringer** [comment]
ok, then we keep this and extend #42 with the secret pin

**Comment by tlaurion** [comment]
>> TLDR: #41>#38>#39, nice to have #40
>#41 was deemed impossible because physical presence removal required nk3 firmware change, and led to #42
> So priorities are:
> #42>#41>#38>#39, nice to have #40
>_Originally posted by @tlaurion in https://github.com/Nitrokey/nitrokey-hotp-verification/issues/36#issuecomment-2506996805_

No need to change a PIN that has no PIN set; if PIN is set but locked, reset secret app PIN needed.
All of which is #42: setting a PIN, not letting it silently be set on first use, possibily with a typo in it.

Can close if #42 implemented. Logic of changing PIN needs 2 PIN by GPG, changing user PIN is from Admin PIN incase of being locked. Nk3 has one single Secure App PIN, even if htp_verification info reports it wrongly today, but that is #38 and #39...

**Comment by tlaurion** [comment]
> ok, then we keep this and extend #42 with the secret pin

**I need #42**. Would love #41 but said not right now. So unattended reownership will be postponed (physical presence still needed) while misleading info reported to users **#38 and #39 would make sense for all support request teams**, including yours.
